File: projects/m2/013-tokenizing-a-string/solutions/lisaHu/solution-013-tokenizing-a-string.py
def tokenizing_string(s):
    #1st list of operators and parenthesis, 2nd list will store result
    ope_par ='[]()+-*/?^'
    tokens = []
    num = ''
    for i in s:
        #if character in string is in list ope_par, it will check if there is something
        #in variable num. if there is, it will append to tokens list and clear the var
        #whether it will append the var num, the character in ope_par will be appended
        if i in ope_par:
            if num:
                tokens.append(num)
                num = ''
            tokens.append(i)
        #unless the character is a blank space, the character will be added to num variable
        #and it will be added only after a character in ope_par will be encountered
        else:
            if i == ' ':
                continue
            num += i
    else:
        #at the end of loop, if the last character of string is not in ope_par, it will be
        #appended in result list
        if num:
            tokens.append(num)
        return tokens

if __name__=='__main__':
    string = input('Enter mathematical expression (no letters):\n')
    print(tokenizing_string(string))

File: projects/m2/013-tokenizing-a-string/solutions/EvaLateltin/solution-013-tokenizing-a-string.js
/*
LOGIC
Create a prompt function that gets an input from the user and store it in a variable
Create an array that will store the tokens
Create a function that: 
- get a string as a parameter
- create a temporary string of numbers
- loop through all characters of the string:
- if a character is a number, add it to the temporary string and keep the loop going. 
In the following iterations, if the following characters are numbers, they will be added to this temporary string as well.
When the number will end, the temporary number will be pushed to the array. 
- if a character is not an empty space, it will be an operator or a parenthesis and it will be pushed to the array.  
Once the loop is concluded, alert the array of tokens.
*/

const enteredString = prompt(
  "Enter a mathematical expression to be tokenized."
);
const tokenArray = [];

function tokenizeString(string) {
  let temporaryString = "";

  for (let i = 0; i < string.length; i++) {
    if (!isNaN(parseInt(string.charAt(i)))) {
      // if the character is a number
      while (!isNaN(parseInt(string.charAt(i)))) {
        temporaryString += string.charAt(i);
        i++;
      }
      tokenArray.push(temporaryString);
      temporaryString = ""; // Reset the temporaryString
      i--; // When the while loop ends, I want to take a step back, otherwise the character right after the last number will not be considered.
    } else if (string.charAt(i) !== " ") {
      // Push the other characters if they are not a blank space
      tokenArray.push(string.charAt(i));
    }
  }
  console.log(tokenArray);
}

tokenizeString(enteredString);
